File: Problem/Easy/1501-1600/1502. Can Make Arithmetic Progression From Sequence/a.java
// To Code : 3 min
// Execution : 2 ms

// Time complexity: O(n) + sort
// Space complexity: O(1)

import java.util.Arrays;

class Solution {
	public boolean canMakeArithmeticProgression(int[] arr) {
		Arrays.sort(arr);
		int diff = arr[1] - arr[0];

		for (int i = 2; i < arr.length; i++)
			if (arr[i] - arr[i - 1] != diff)
				return false;

		return true;
	}
}

File: Problem/Easy/1501-1600/1502. Can Make Arithmetic Progression From Sequence/_a.py
class Solution:
    def canMakeArithmeticProgression(self, arr) -> bool:
        arr.sort()
        diff = arr[1] - arr[0]

        for i in range(1, len(arr)):
            if (arr[i] - arr[i-1] != diff):
                return False
        # Note:- else of for loop
        else:
            return True
